### PR TITLE
Shapes up ServerCommandLineArgs parsing

### PR DESCRIPTION
**Pull request by tinwelint**
The format was undocumented and untested. Also the integration tests that
used these features specified configuration options inconsistently.

### CODE REVIEW

File: community/server/src/test/java/org/neo4j/server/ServerCommandLineArgsTest.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                Pair.of( "my_second_option", "second" ) ),
+
+                asSet( parsed.configOverrides() ) );
+    }
+
+    // Silly java
+    private String[] args( String... args )
+    {
+        return args;
+    }
```
Comment (davidegrohmann) [inline community/server/src/test/java/org/neo4j/server/ServerCommandLineArgsTest.java]: perhaps move this method to `ArrayUtil`?

**Comment by davidegrohmann** [comment]
@tinwelint there is test failure to be fixed

File: community/server/src/test/java/org/neo4j/server/BaseBootstrapperTest.java [hunk 2]
```diff
@@ -92,26 +93,36 @@ public void canOverrideConfigValues() throws Throwable
...
         File configFile = tempDir.newFile( "neo4j.config" );
 
-        MapUtil.store( stringMap(
+        store( stringMap(
                 forced_kernel_id.name(), "thisshouldnotshowup"
         ), configFile );
 
         // When
-        start( bootstrapper, new String[]{
+        start( bootstrapper, args(
```
Comment (davidegrohmann) [inline community/server/src/test/java/org/neo4j/server/BaseBootstrapperTest.java]: `ArrayUtil.array(...)` instead?

**Comment by tinwelint** [comment]
@davidegrohmann Can we get this in asap, so that no more builds break because of it?

**Pull request merged**